### PR TITLE
fix: crash loading non-standard schemes in iframes

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/35462.
Closes https://github.com/electron/electron/issues/28407.

This fixes a crash that occurs when loading non-standard schemes from iframes. This was happening because
`ChildProcessSecurityPolicyImpl::CanAccessDataForOrigin` contains explicit exceptions to allow built-in non-standard schemes, but does not check for non-standard schemes registered by the embedder.

Upstream, CRBUG:1081397 contains several paths forward - here I chose to swap out the`CHECK` in `navigation_request.cc` from `policy->CanAccessDataForOrigin` to`policy->CanCommitOriginAndUrl`.

CL opened at CL:3856266

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed a crash that occured when loading non-standard schemes from iframes.

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]

File: patches/chromium/fix_crash_loading_non-standard_schemes_in_iframes.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ChildProcessSecurityPolicyImpl::CanAccessDataForOrigin contains explicit
+exceptions to allow built-in non-standard schemes, but does not check
+for non-standard schemes registered by the embedder.
+
+Upstream, https://bugs.chromium.org/p/chromium/issues/detail?id=1081397
+contains several paths forward - here I chose to swap out the
+CHECK in navigation_request.cc from policy->CanAccessDataForOrigin to
+policy->CanCommitOriginAndUrl.
+
+This, or a version of it, will be upstreamed.
```
Comment (ckerr) [inline patches/chromium/fix_crash_loading_non-standard_schemes_in_iframes.patch:17]: Is there a link for the upstream code submission?

Comment (codebytere) [reply patches/chromium/fix_crash_loading_non-standard_schemes_in_iframes.patch:17]: https://chromium-review.googlesource.com/c/chromium/src/+/3856266!

**Comment by george-thomas-hill** [comment]
Thank you for doing this!

**Comment by george-thomas-hill** [comment]
@codebytere :

Update:

I'm still finding (in Electron 21.2.0) that loading a PDF in an iFrame fails to display the PDF, and loading a PDF in a webview crashes Electron.

Here's an Electron Fiddle gist that demonstrates this:

https://gist.github.com/george-thomas-hill/c3179e6eddeb1a2715201a20dbf95015

Could you please take a look?

**Pull request merged**